pkg/cosign/verify.go

```diff
@@ -675,27 +654,65 @@ func verifyInternal(ctx context.Context, sig oci.Signature, h v1.Hash,
 		}
 	}
 
+	// 2. Check the validity time of the signature.
+	// This is the signature creation time. As a default upper bound, use the current
+	// time.
+	validityTime := time.Now()
+
 	bundleVerified, err = VerifyBundle(ctx, sig, co.RekorClient)
 	if err != nil {
 		return false, fmt.Errorf("error verifying bundle: %w", err)
 	}
 
-	// If the --offline flag was specified, fail here
+	if bundleVerified {
+		// Update with the verified bundle's integrated time.
+		validityTime, err = getBundleIntegratedTime(sig)
+	}
+
+	// If the --offline flag was specified, fail here. bundleVerified returns false with
+	// no error when there was no bundle provided.
+	// TODO: You can be offline when you are verifying with a public key. This shouldn't
```

Comment:
@priyawadhwa WDYT?

Reply:
I think it's solved by the flag that says "don't check the tlog", and it should remain here for now
This is an error saying "Signature has not been checked against a tlog"